### DISCUSSION TITLE
Why doesn't demo codes use function generators on async actions

### DISCUSSION THREAD

**Discussion by osrl** [infinitered/ignite#2295] [Q&A] [answered]
Hi, 

As stated in [mst documentation](https://mobx-state-tree.js.org/concepts/async-actions#using-generators) using function generators, `flow` and `yield` is the recommended method of using async actions on MST.

Why doesn't demo codes use function generators on async actions? Am I missing something? I've tried to search through pull requests but no luck.

**Comment by jamonholmgren** [comment, marked as answer]
For a long time I didn't really agree with the documentation and felt like `async` was better, so most of the docs / examples I'd write would use async. However, after running into some issues with `async` this year, I've since come around and do think that `yield`/`flow` is indeed better. So we should align the examples with that recommendation. Good catch @osrl.

**Reply by osrl** [reply]
thanks @jamonholmgren. Also thank you for the great work. I've used ignite-bowser 2 or 3 years ago. A lot have changed since that time. No more extensions, no more Provider etc. It looks great.
